Confusing error if ipynb-filters are used

## Followed tutorial but getting error

Followed the [nbdev tutorial](https://nbdev.fast.ai/tutorials/tutorial.html) for nbdev_cards example and did `git push` results.  but I keep getting errors about CI and Github pages failing because quarto needs root access.

Also  ran `nbdev_clean` but the problem persists

![installing quarto requires root](https://user-images.githubusercontent.com/61335136/219520895-995855e6-51ea-40d9-b925-5fc8fe02de6a.png)


## Minimal reproducible example

https://github.com/rleyvasal/nbdev_cards

The error is a not an permission error about root access.

It is `AttributeError: path_`. From your most recent action I have downloaded the following log:

```sh
Quarto is not installed. We will download and install it for you.
Installing or upgrading quarto -- this requires root access.
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.16/x64/bin/nbdev_filter", line 8, in <module>
    sys.exit(nbdev_filter())
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/fastcore/script.py", line 119, in _f
    return tfunc(**merge(args, args_from_prog(func, xtra)))
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/nbdev/cli.py", line 46, in nbdev_filter
    with redirect_stdout(dn): filt(nb)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/nbdev/processors.py", line 243, in __call__
    def __call__(self, nb): return self.nb_proc(nb).process()
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/nbdev/process.py", line 126, in process
    for proc in self.procs: self._proc(proc)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/nbdev/process.py", line 120, in _proc
    if hasattr(proc,'end'): proc.end()
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/nbdev/frontmatter.py", line 66, in end
    self.fm.update({'output-file': _nbpath2html(Path(self.nb.path_)).name})
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/fastcore/basics.py", line 247, in __getattr__
    def __getattr__(self,k): return self[k] if k in self else stop(AttributeError(k))
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/fastcore/basics.py", line 216, in stop
    raise e
AttributeError: path_
Error: Process completed with exit code 1.
```

I had a look at https://github.com/fastai/workflows and didn't see any recent commits that may have broke things. So I followed the tutorial with https://github.com/tourdownunder/nbdev-hello-world and don't get this issue. 

It may have been intermittent. What happens if you run it again?

--- a/nbdev/frontmatter.py
+++ b/nbdev/frontmatter.py
@@ -63,5 +63,10 @@ class FrontmatterProc(Processor):
     def end(self):
         self.nb.frontmatter_ = self.fm
         if not self.fm: return
+        if not hasattr(self.nb, 'path_'):
+            raise AttributeError('Notebook missing `path_` attribute.\n\nPlease remove any nbdev-related notebook filters '
+                                 'from your _quarto.yml file (e.g. `ipynb-filter: [nbdev_filter]`), since they are no '
+                                 'longer supported as of nbdev v2.3. See the v2.3 launch post for more information: '
+                                 'https://forums.fast.ai/t/upcoming-changes-in-v2-3-edit-now-released/98905.')
         self.fm.update({'output-file': _nbpath2html(Path(self.nb.path_)).name})
         _insertfm(self.nb, self.fm)

--- a/nbs/api/09_frontmatter.ipynb
+++ b/nbs/api/09_frontmatter.ipynb
@@ -126,6 +126,11 @@
     "    def end(self):\n",
     "        self.nb.frontmatter_ = self.fm\n",
     "        if not self.fm: return\n",
+    "        if not hasattr(self.nb, 'path_'):\n",
+    "            raise AttributeError('Notebook missing `path_` attribute.\\n\\nPlease remove any nbdev-related notebook filters '\n",
+    "                                 'from your _quarto.yml file (e.g. `ipynb-filter: [nbdev_filter]`), since they are no '\n",
+    "                                 'longer supported as of nbdev v2.3. See the v2.3 launch post for more information: '\n",
+    "                                 'https://forums.fast.ai/t/upcoming-changes-in-v2-3-edit-now-released/98905.')\n",
     "        self.fm.update({'output-file': _nbpath2html(Path(self.nb.path_)).name})\n",
     "        _insertfm(self.nb, self.fm)"
    ]

